src/WLDGrant.sol

```diff
@@ -10,7 +10,10 @@ contract WLDGrant is IGrant {
         if (timestamp < launchDayTimestampInSeconds) revert InvalidGrant();
 
         uint256 weeksSinceLaunch = (timestamp - launchDayTimestampInSeconds) / 1 weeks;
-        return 15 + (weeksSinceLaunch - 3) / 2;
+        uint256 grantId = 15 + (weeksSinceLaunch - 3) / 2;
+        // Grant 29 is a four-week grant.
+        if (grantId <= 29) return grantId;
+        return grantId - 1;
```

Comment:
Makes sense. In some future version we can fold it into the offset on line 13.

Reply:
Yes, we can do this if we don't use the contract for past reservations (i.e. with the separate contracts as we have currently).